clients/admin-ui/src/features/common/table/v2/FidesTable.tsx

```diff
@@ -74,19 +75,20 @@ const tableHeaderButtonStyles = {
   },
 };
 
+interface HeaderContentProps<T> {
```

Comment:
Question for you, is there a significant difference between TS interfaces and types for component props?

Reply:
Great question.

Yes and No. The general rule is "use `interface` until you need `type`". Not a very big difference, especially if you aren't exporting it. If exporting, you run the risk of 2 interfaces with the same name conflicting (extending) one another accidentally, where that's not possible with type.

For example:
```
interface Props {
  prop1: boolean;
}
interface Props {
  prop2: string;
} 
```
creates the equivalent of
```
interface Props {
  prop1: boolean;
  prop2: string;
}
```
if those were both using `type` then only `prop2` would apply.

I think the main reason our app is inundated with `type` is because there is no naming convention and just plain old `Props` is being used everywhere, which was likely creating the scenario where they were accidentally extending each other.

I prefer `interface` for components because a) if you are consistent with naming conventions (eg. `${componentName}Props`) and never export them unless absolutely necessary (rare) then the general rule "use `interface` until you need `type`" applies. b) this is what most component libraries are using (see Chakra's components, eg. `BoxProps`) so extending those with like definitions feels cleaner. c) nitpicky: it's more clear to use `extends` than `&` IMO.